pkgdb/src/flox-flake.cc

```diff
@@ -65,29 +77,64 @@ FloxFlake::openEvalCache()
 {
   if ( this->_cache == nullptr )
     {
-      auto fingerprint = this->lockedFlake.getFingerprint();
-      this->_cache     = std::make_shared<nix::eval_cache::EvalCache>(
-        ( nix::evalSettings.useEvalCache && nix::evalSettings.pureEval )
-              ? std::optional { std::cref( fingerprint ) }
-              : std::nullopt,
-        *this->state,
-        [&]()
-        {
-          nix::Value * vFlake = this->state->allocValue();
-          nix::flake::callFlake( *this->state, this->lockedFlake, *vFlake );
-          this->state->forceAttrs( *vFlake,
-                                   nix::noPos,
-                                   "while parsing cached flake data" );
-          nix::Attr * aOutputs
-            = vFlake->attrs->get( this->state->symbols.create( "outputs" ) );
-          assert( aOutputs != nullptr );
-          return aOutputs->value;
-        } );
+      this->_cache = flox::openEvalCache( *this->state, this->lockedFlake );
     }
   return static_cast<nix::ref<nix::eval_cache::EvalCache>>( this->_cache );
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+
+nix::Value *
+flakeLoader( nix::EvalState &                state,
+             const nix::flake::LockedFlake & lockedFlake )
+{
+  nix::Value * vFlake = state.allocValue();
+  /* Evaluate the `outputs' function using `inputs' as arguments. */
+  nix::flake::callFlake( state, lockedFlake, *vFlake );
+  state.forceAttrs( *vFlake, nix::noPos, "while parsing cached flake data" );
+  /* Overwrite the _global_ `outputs` symbol with the evaluated result.
+   * This makes the original `outputs` function inaccessible. */
+  nix::Attr * aOutputs
+    = vFlake->attrs->get( state.symbols.create( "outputs" ) );
+  assert( aOutputs != nullptr );
+  return aOutputs->value;
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+/** @brief Open a `nix::eval_cache::EvalCache` for a locked flake. */
+nix::ref<nix::eval_cache::EvalCache>
+openEvalCache( nix::EvalState &                state,
+               const nix::flake::LockedFlake & lockedFlake )
+{
+  auto fingerprint = lockedFlake.getFingerprint();
+  auto useCache = std::make_optional<std::reference_wrapper<const nix::Hash>>(
+    std::cref( fingerprint ) );
+
+  /* Push current settings. */
+  bool oldUseCache = nix::evalSettings.useEvalCache;
+  bool oldPureEval = nix::evalSettings.pureEval;
+
+  nix::evalSettings.useEvalCache.assign( true );
+  nix::evalSettings.pureEval.assign( true );
+
+  /* Loads a flake into the `nix` evaluator and a SQLite3 cache database. */
+  auto cache = std::make_shared<nix::eval_cache::EvalCache>(
+    useCache,
+    state,
+    [&state, &lockedFlake]() { return flakeLoader( state, lockedFlake ); } );
+
+  /* Pop old settings. */
```

Comment:
```suggestion
  /* Restore old settings. */
```

Reply:
This comment and the one are a push/pop pair. I updated it to just say "save"/"restore" instead.